### PR TITLE
YM-446 | Add audit logging for session and token based authentication

### PR DESCRIPTION
**Pull request by charn**
Current audit logging implementation is extended to include logs for session and token based authentication events. The format for these new logs should be similar to existing audit log events. Audit log format is also updated to include e.g. the IP of the calling client.

### CODE REVIEW

File: common_utils/audit_logging.py [hunk 1]
```diff
@@ -36,39 +61,61 @@ def log(action, instance):
...
-        message["audit_event"]["target"]["user_id"] = (
-            str(target_user.uuid) if hasattr(target_user, "uuid") else None
-        )
-        if settings.AUDIT_LOG_USERNAME:
-            message["audit_event"]["target"]["user_name"] = (
-                target_user.username if hasattr(target_user, "username") else None
-            )
+    ip_address = get_original_client_ip()
+    if ip_address:
+        message["audit_event"]["jassaribe"] = {
```
Comment (klempine) [inline common_utils/audit_logging.py:85]: What's the reasoning behind naming the extra field as `jassaribe`? Would this mean that if Berth did something similar with their logging, they would add a new extra field as `venebe`? If an admin would want to look for all log entries done from the same IP address, they would have to make sure to include all possible backends to their search, right?

Would this be better if we named it as something more generic? Like `extra_info` or something?

**Review by klempine — COMMENTED** [review]
This looks good, but I'd like some clarification on the one comment I made before approving.

File: common_utils/audit_logging.py (hunk 1 above)
Comment (charn) [reply common_utils/audit_logging.py:85]: I basically just mimicked what was done in open-city-profile where this field was `profilebe`, but someting like `extra_info` does sound quite reasonable. :+1:

**Review by klempine — APPROVED** [review]
🥇

**Pull request merged**